ui/src/app/cohort-search/list-overview/list-overview.component.tsx

```diff
@@ -0,0 +1,426 @@
+      switch (action) {
+        case 'notebook':
+          url += 'notebooks';
+          break;
+        case 'review':
+          url += `cohorts/${cohort.id}/review`;
+          break;
+      }
+      navigateByUrl(url);
+    }
+
+    toggleChartMode() {
+      const {stackChart} = this.state;
+      this.setState({stackChart: !stackChart});
+    }
+
+    render() {
+      const {cohort, chartData, deleting, error, loading, saveModal, name, description, nameTouched,
+        saving, saveError, stackChart, total} = this.state;
+      const disableIcon = loading || !cohort ;
+      const disableSave = cohort && cohort.criteria === this.criteria;
+      const invalid = nameTouched && !name;
+      const items = [
+        {label: 'Save', command: () => this.saveCohort(), disabled: disableSave},
+        {label: 'Save as', command: () => this.setState({saveModal: true})},
+      ];
+      return <React.Fragment>
+        <div>
+          <div style={styles.overviewHeader}>
+            <div style={{width: '100%'}}>
+              {!!cohort && <React.Fragment>
+                <Menu appendTo={document.body}
+                  model={items} popup={true} ref={el => this.dropdown = el} />
+                <Button type='primary' style={styles.saveButton}
+                  onClick={(event) => this.dropdown.toggle(event)}
+                  disabled={loading || saving || this.hasErrors}>
+                  SAVE COHORT <ClrIcon shape='caret down' />
+                </Button>
+              </React.Fragment>}
+              {!cohort && <Button type='primary'
+                onClick={() => this.setState({saveModal: true})}
+                style={styles.saveButton}
+                disabled={loading || this.hasErrors}>CREATE COHORT</Button>}
+              <TooltipTrigger content={<div>Export to notebook</div>}>
+                <Clickable style={{...styles.actionIcon, ...styles.disabled}}
+                  onClick={() => this.navigateTo('notebook')} disabled>
+                  <ClrIcon shape='export' className='is-solid' size={30} />
+                </Clickable>
+              </TooltipTrigger>
+              <TooltipTrigger content={<div>Delete cohort</div>}>
+                <Clickable style={{...styles.actionIcon, ...(disableIcon ? styles.disabled : {})}}
+                  onClick={() => this.setState({deleting: true})}>
+                  <ClrIcon shape='trash' className='is-solid' size={30} />
+                </Clickable>
+              </TooltipTrigger>
+              <TooltipTrigger content={<div>Review participant level data</div>}>
+                <Clickable style={{...styles.actionIcon, ...(disableIcon ? styles.disabled : {})}}
+                  onClick={() => this.navigateTo('review')}>
+                  <ClrIcon shape='copy' className='is-solid' size={30} />
+                </Clickable>
+              </TooltipTrigger>
+            </div>
+            <h2 style={styles.totalCount}>
+              Total Count: &nbsp;
```

Comment:
I would change this logic to 
```
{this.hasErrors ? <span>error tooltip</span> : 
   {loading ? <Spinner /> : <span> totals count</span>}
}
```
more concise and avoids having to keep checking the error logic.  can you move the check of if total is undefined into `this.hasErrors`?   and what is the actual case where that will happen, if neither of the other error conditions are met?

Reply:
You're right, much cleaner this way. I had the undefined check on total because I was getting errors on component init, but I changed the initial value of loading to true and that error goes away.